typesafety/test_records.yml

```diff
@@ -0,0 +1,76 @@
+---
+- case: no_error_true
+  parametrized:
+    - level: info
+    - level: debug
+    - level: error
+    - level: trace
+  main: |
+    from loguru import logger
+
+    LOG = logger.opt(record=True)
+
+    LOG.{{ level }}('test')
+    LOG.{{ level }}('r={record["line"]}')
+    LOG.{{ level }}('{record["line"]}')
+  out: |
+    main:5: note: Logger configured with record=True is not using record structure
+- case: no_error_false
+  main: |
+    from loguru import logger
+
+    logger.opt(record=False).info('test')
+- case: record_in_kwargs
+  parametrized:
+    - record: "True"
+      out: "main:4: error: record keyword argument cannot override record structure  [logger-kwarg]"
+    - record: "False"
+      out: "main:4: error: Logger is accessing record structure without record=True  [logger-record]"
+  main: |
+    from loguru import logger
+
+    logger.opt(record={{ record  }}).info(
+      'line is {record["line"]}',
+      record={
+        'line': 1,
+      },
+    )
+  out: "{{ out }}"
+- case: record_keys_errors
+  parametrized:
+    - key: foo
+    - key: elapsedd
+    - key: odin
+  main: |
+    from loguru import logger
+
+    logger.opt(record=True).info(
+      'Accessing {record["{{ key }}"]} makes for error={yes_no}',
+      yes_no=len('{{ out }}') != 0,
+    )
+  out: |
+    main:4: error: Logger record structure does not contain {{ key }} key  [logger-record]
+- case: record_keys_ok
+  parametrized:
+    - key: line
+    - key: exception
+    - key: elapsed
+    - key: exception
```

Comment:
The `exception` looks already parametrized two lines above. ;)

Reply:
dang it! perhaps I should implement another patch to `pytest-mypy-plugin`. What do you think @sobolevn? I guess it kinda makes sense to detect parameters duplicates? Or maybe not - pytest does not seem to act like that by default :|. Anyhoo :D thx for noticing that :D

Reply:
Yes! Good idea!

Reply:
Duplicates removed and issue for `pytest-mypy-plugins` already created.